zcash_client_sqlite/src/wallet/init/migrations/addresses_table.rs

```diff
@@ -64,9 +62,8 @@ impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
 
         let mut rows = stmt_fetch_accounts.query([])?;
         while let Some(row) = rows.next()? {
-            let account: u32 = row.get(0)?;
-            let account = AccountId::try_from(account).map_err(|_| {
-                WalletMigrationError::CorruptedData("Account ID is invalid".to_owned())
+            let account = AccountId::try_from(row.get::<_, u32>(0)?).map_err(|_| {
+                WalletMigrationError::CorruptedData("Invalid ZIP-32 account index.".to_string())
```

Comment:
```suggestion
                WalletMigrationError::CorruptedData("Invalid ZIP-32 account index.".to_owned())
```

Reply:
I'd love to learn the significance of `to_owned` vs `to_string` for this.